armadillo/src/main/java/org/molgenis/armadillo/storage/ArmadilloWorkspace.java

```diff
@@ -1,13 +1,13 @@
 package org.molgenis.armadillo.storage;
 
 import java.io.ByteArrayInputStream;
-import java.io.IOException;
 import java.io.InputStream;
 import org.apache.commons.io.IOUtils;
 import org.molgenis.armadillo.exceptions.StorageException;
 
 public class ArmadilloWorkspace {
   byte[] content;
+  public static final String WORKSPACE_TOO_BIG_ERROR = "Unable to load workspace. Maximum supported workspace size is 2GB";
 
   public ArmadilloWorkspace(InputStream is) {
     content = toByteArray(is);
```

Comment:
Can we instead do a while(read(10000 bytes){count++} and then not having to load all in memory but still count?